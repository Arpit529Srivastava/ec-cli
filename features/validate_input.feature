Feature: validate input
  The ec command line should be able to inspect input files
  
  Background:
    Given stub git daemon running

  Scenario: valid policy URL
    Given a git repository named "happy-day-config" with
      | policy.yaml | examples/happy_config.yaml |
    Given a git repository named "happy-day-policy" with
      | main.rego | examples/happy_day.rego |
    Given a pipeline definition file named "pipeline_definition.yaml" containing
    """
    ---
    apiVersion: tekton.dev/v1
    kind: Pipeline
    metadata:
      name: basic-build
    spec:
      tasks:
      - name: appstudio-init
        taskRef:
          name: init
          version: "0.1"
    """
    When ec command is run with "validate input --file pipeline_definition.yaml --policy git::https://${GITHOST}/git/happy-day-config.git"
    Then the exit status should be 0
    Then the output should match the snapshot

  Scenario: policy URL with no rego files
    Given a git repository named "sad-day-config" with
      | policy.yaml | examples/sad_config.yaml |
    Given a git repository named "sad-day-policy" with
      | main.reg0 | examples/happy_day.rego |
    Given a pipeline definition file named "pipeline_definition.yaml" containing
    """
    ---
    apiVersion: tekton.dev/v1
    kind: Pipeline
    metadata:
      name: basic-build
    spec:
      tasks:
      - name: appstudio-init
        taskRef:
          name: init
          version: "0.1"
    """
    When ec command is run with "validate input --file pipeline_definition.yaml --policy git::https://${GITHOST}/git/sad-day-config.git"
    Then the exit status should be 1
    Then the output should match the snapshot
