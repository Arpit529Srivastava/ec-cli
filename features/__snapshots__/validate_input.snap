
[invalid policy URL:stdout - 1]

---

[invalid policy URL:stderr - 1]
Error: 1 error occurred:
    * error validating file pipeline_definition.yaml: no rego files found in policy subdirectory



---

[valid policy URL:stdout - 1]
{
  "success": true,
  "filepaths": [
    {
      "filepath": "pipeline_definition.yaml",
      "violations": [],
      "warnings": [],
      "successes": null,
      "success": true,
      "success-count": 1
    }
  ],
  "policy": {
    "sources": [
      {
        "policy": [
          "git::https://${GITHOST}/git/happy-day-policy.git"
        ]
      }
    ]
  },
  "ec-version": "${EC_VERSION}",
  "effective-time": "${TIMESTAMP}"
}
---

[valid policy URL:stderr - 1]

---

[policy URL with no rego files:stdout - 1]

---

[policy URL with no rego files:stderr - 1]
Error: 1 error occurred:
    * error validating file pipeline_definition.yaml: no rego files found in policy subdirectory



---
